react/features/app/actions.js

```diff
@@ -25,6 +25,48 @@ export function appNavigate(uri: ?string) {
         _appNavigateToOptionalLocation(dispatch, getState, parseURIString(uri));
 }
 
+/**
+ * Redirects to another page generated by replacing the path in the original URL
+ * with the given path.
+ *
+ * @param {(string)} pathname - The path to navigate to.
+ * @returns {Function}
+ */
+export function redirectWithStoredParams(pathname: string) {
+    return (dispatch: Dispatch<*>, getState: Function) => {
+        const { locationURL } = getState()['features/base/connection'];
+        const newLocationURL = new URL(locationURL.href);
+
+        newLocationURL.pathname = pathname;
+        window.location.assign(newLocationURL.toString());
+    };
+}
+
+/**
+ * Reloads the page by restoring the original URL.
+ *
+ * @returns {Function}
+ */
+export function reloadWithStoredParams() {
+    return (dispatch: Dispatch<*>, getState: Function) => {
+        const { locationURL } = getState()['features/base/connection'];
+        const windowLocation = window.location;
+        const oldSearchString = windowLocation.search;
+
+        windowLocation.replace(locationURL.toString());
+
+        if (window.self !== window.top
+            && locationURL.search === oldSearchString) {
+            // NOTE: Assuming that only the hash or search part of the URL will
+            // be changed!
+            // location.reload will not trigger redirect/reload for iframe when
+            // only the hash params are changed. That's why we need to call
+            // reload in addition to replace.
+            windowLocation.reload();
+        }
+    };
```

Comment:
Before we were doing replace or reload and now the logic can go replace and reload is this ok?

Reply:
Yes. Actually this is the part that solves the problem. 

Replace will change the URL to the original one but if there are only hash params, this won't trigger reload. And the reload call after that will reload the page with the correct params. 

But if there are search params (jwt token for example) replace will do the whole job and we won't call reload.

Reply:
I see, thanks.